tmail-backend/blob/blobid-list/src/test/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreContract.scala

```diff
@@ -0,0 +1,52 @@
+package com.linagora.tmail.blob.blobid.list
+
+import SingleSaveBlobStoreDAOFixture._
+import org.apache.james.blob.api.{BlobId, BlobStoreDAO, ObjectStoreException}
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable
+import org.junit.jupiter.api.Test
+import reactor.core.scala.publisher.SMono
+
+trait SingleSaveBlobStoreContract {
+
+  def blobStoreDAO: BlobStoreDAO
+
+  def blobIdList: BlobIdList
+
+  def blobIdFactory: BlobId.Factory
+
+  @Test
+  def saveShouldSuccessBeIfBlobIsNotExits(): Unit = {
+    val blobId: BlobId = blobIdFactory.randomId()
+    SMono.fromPublisher(blobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block()
+    val isStore: Boolean = SMono.fromPublisher(blobIdList.isStored(blobId)).block()
+    assert(isStore)
+  }
+
+  @Test
+  def saveShouldBeFailIfBlobIsExits(): Unit = {
+    val blobId: BlobId = blobIdFactory.randomId()
+    SMono.fromPublisher(blobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block()
+
+    val duplicateBlobThrowingCallable: ThrowingCallable = () =>
+      SMono.fromPublisher(blobStoreDAO.save(TEST_BUCKET_NAME, blobId, SHORT_BYTEARRAY)).block()
+
+    assertThatThrownBy(duplicateBlobThrowingCallable)
+      .isInstanceOf(classOf[ObjectStoreException])
+    assertThatThrownBy(duplicateBlobThrowingCallable)
+      .hasMessage("Can not save duplicate blob when single save is enable")
+
+  }
+
+  @Test
+  def deleteShouldBeFail(): Unit = {
```

Comment:
```suggestion
  def deleteShouldFail(): Unit = {
```

Reply:
You try to delete a blobId that does not exist here. How about trying to delete a blobId that has been stored previously? That test is missing